unit_tests/testParameterLists/testParameterLists.f90

```diff
@@ -4807,6 +4808,58 @@ SUBROUTINE testHas()
     CALL clear_test_vars()
   ENDSUBROUTINE testHas
 !
+!-------------------------------------------------------------------------------
+  SUBROUTINE testConvertTo2DStringArray()
+    INTEGER(SIK) :: i,j
+    TYPE(StringType) :: addr
+    TYPE(StringType),ALLOCATABLE :: table(:,:),reftable(:,:)
+
+    !Test null call
+    CALL testParam%clear()
+    addr=''
+    CALL testParam%convertTo2DStringArray(addr,table)
+    ASSERT(.NOT. ALLOCATED(table),'empty call')
+
+    !Setup the PL
+    CALL testParam%add('TestPL->List1->1->Row1','Row1')
+    CALL testParam%add('TestPL->List1->1->testRow2','testRow2')
+    CALL testParam%add('TestPL->List1->1->testRow3; Extra','testRow3; Extra')
+    CALL testParam%add('TestPL->List1->2->Row1',10)
+    CALL testParam%add('TestPL->List1->2->testRow2',1000000)
+    CALL testParam%add('TestPL->List1->2->testRow3; Extra',23.0E+10_SRK)
+    CALL testParam%add('TestPL->List1->3->Row1',.TRUE.)
+    CALL testParam%add('TestPL->List1->3->testRow2',.FALSE.)
+    CALL testParam%add('TestPL->List1->3->testRow3; Extra','23.0E+10')
+    CALL testParam%add('TestPL->List1->4->Row1','TRUE')
+    CALL testParam%add('TestPL->List1->4->testRow2','FALSE')
+    CALL testParam%add('TestPL->List1->4->testRow3; Extra','-')
```

Comment:
Should there be an entry in the unit test for a derived type to ensure that the conversion terminates 'gracefully' (i.e. no segfaults) in this case? Or is the assumption that this kind of function call will never happen?

Reply:
There is no corresponding `%add` call in the generic interface for derived types, so that is assumed to never happen.  But, analogous to your point, I should ensure that the test does in fact test all the intrinsic kinds we do support.  I'll expand the test accordingly.

Reply:
Ok, updated the test to have all supported `paramtypes` tested.